river/cluster/textclust.py

```diff
@@ -0,0 +1,595 @@
+        self.microToMacro = None
+        self.upToDate = False
+        self.realtime = None
+        self.dist_mean = 0
+
+        # create a new distance instance for micro and macro distances.
+        self.micro_distance: textclust.distances = self.distances(micro_distance)
+        self.macro_distance: textclust.distances = self.distances(macro_distance)
+
+    def learn_one(self, x, time=None, sample_weight=None, **kwargs):
+
+        ## if id is specified
+        if kwargs["id"]:
+            id = kwargs["id"]
+        else:
+            id = None
+
+        localdict = {}
+        for key in x.keys():
+            new_key = key
+            localdict[new_key] = {}
+            localdict[new_key]["tf"] = x[key]
+            localdict[new_key]["ids"] = [id]
+        ngrams = localdict
+        ngrams = dict(ngrams)
+
+        # set up to date variable. it is set when everything is faded
+        self.upToDate = False
+
+        # check if realtime fading is on or not. specify current time accordingly
+        if self.realtimefading:
+            self.t = time
+        else:
+            self.t = self.n
+
+        # realtime is only the current time non decoded to store for the plotter
+        if self.realtime is not None:
+            self.realtime = self.realtime
+
+        # if there is something to process
+        if len(ngrams) > 0:
+
+            # create artificial micro cluster with one observation
+            mc = self.microcluster(ngrams, self.t, 1, self.realtime, id, self.clusterId)
+
+            # calculate idf
+            idf = self._calculateIDF(self.microclusters.values())
+
+            clusterId, min_dist = self._get_closest_mc(mc, idf, self.micro_distance)
+
+            # if we found a cluster that is close enough we merge our incoming data into it
+            if clusterId is not None:
+
+                self.num_merged_obs += 1
+                ## add number of observations
+                self.microclusters[clusterId].n += 1
+
+                self.microclusters[clusterId].merge(
+                    mc, self.t, self.omega, self._lambda, self.termfading, self.realtime
+                )
+                self.assignment[self.n] = clusterId
+                self.dist_mean += min_dist
+
+            # if no close cluster is found we check if embeddings should be verified and maybe create a new one
+            else:
+                self.dist_mean += min_dist
+                clusterId = self.clusterId
+                self.assignment[self.n] = clusterId
+                self.microclusters[clusterId] = mc
+                self.clusterId += 1
+        else:
+            print("error - nothing to process")
+
+        # cleanup every tgap
+        if self.lastCleanup is None or self.t - self.lastCleanup >= self.tgap:
+            self._cleanup()
+
+        ## increment observation counter
+        self.n += 1
+        return clusterId
+
+    def predict_one(self, x, sample_weight=None, type="micro"):
```

Comment:
You have to document the `type` parameter here. Also wouldn't it be preferable to have this in the `__init__`?